test/wasm2js/fac.2asm.js

```diff
@@ -17,67 +17,61 @@ function asmFunc(imports) {
  function $0($0_1, $0$hi) {
   $0_1 = $0_1 | 0;
   $0$hi = $0$hi | 0;
-  var i64toi32_i32$3 = 0, i64toi32_i32$5 = 0, i64toi32_i32$2 = 0, i64toi32_i32$1 = 0, $12 = 0, $12$hi = 0, $8 = 0, $8$hi = 0;
-  i64toi32_i32$2 = $0_1;
+  var i64toi32_i32$3 = 0, i64toi32_i32$5 = 0, i64toi32_i32$1 = 0, $8 = 0, $8$hi = 0, $6 = 0, $6$hi = 0;
   i64toi32_i32$1 = 0;
   i64toi32_i32$3 = 0;
-  if ((i64toi32_i32$2 | 0) == (i64toi32_i32$3 | 0) & ($0$hi | 0) == (i64toi32_i32$1 | 0) | 0) {
-   i64toi32_i32$2 = 0;
-   $12 = 1;
-   $12$hi = i64toi32_i32$2;
+  if (($0_1 | 0) == (i64toi32_i32$3 | 0) & ($0$hi | 0) == (i64toi32_i32$1 | 0) | 0) {
+   $8 = 1;
+   $8$hi = 0;
```

Comment:
I am surprised to see codegen changes here. This change even looks non-NFC. Do you know why this changed?

Reply:
It looks like the source file contains `then` and `else` blocks containing single instructions. Those were previously parsed as singleton blocks, but now they're parsed as the actual instructions. I think that change to the IR probably explains the different output here.

Reply:
Could be. But looking at the output it's not obvious to me if it is or isn't a regression (line 21 is gone, for example - these aren't just renamed variables, there is a structural change), so I think it's worth investigating. Let me know if you'd like help with that.

Reply:
Ok, I verified that the different parsing behavior explains this. I ran wasm2js on one of the source functions with and without the special case for singleton `then` and `else` blocks and the output had precisely these differences.

Reply:
Great, thanks!